tests/parser.js

```diff
@@ -148,24 +148,29 @@ define([
 
         it("should set required correctly", function() {
             util.loadXML(REQUIRED_XML);
-
             var q1 = util.getMug("question1");
-            assert(q1.p.required = "true()");
-            assert(q1.p.requiredCondition = "true()");
-            assert(q1.messages.get("requiredAttr").length === 0);
-            assert(q1.messages.get("requiredCondition").length === 0);
+            assert(q1.p.requiredAttr, "required is false")
+            assert.equal(q1.p.requiredCondition, undefined);
+            assert(q1.messages.get("requiredAttr").length === 0,
+                   "requiredAttr has " + q1.messages.get("requiredAttr").length + " messages");
```

Comment:
Nit: curious why you didn't use `assert.equal`?
```js
assert.equal(q1.messages.get("requiredAttr").length, 0, "requiredAttr");
```
Note the third argument can be used for disambiguation when there are multiple asserts in the same test.

It would generate failure messages like

```
The parser should set required correctly
  requiredAttr: expected 1 to equal 0
```

Even better:
```js
assert.deepEqual(q1.messages.get("requiredAttr"), [], "requiredAttr");
```
would show the messages if there is a failure. This can be very useful for debugging test failures.

Reply:
yeah, as you note here it's mainly because of the message generated. it seemed clearer than just some arbitrary `expected 1 to equal 0`. I hadn't tried out a third arg for assert.equals though. I'll add something like that